android/src/play/java/com/dooboolab/RNIap/RNIapModule.kt

```diff
@@ -100,20 +97,21 @@ class RNIapModule(
             promise.safeResolve(true)
             return
         }
-        billingClientCache = builder.setListener(this).build()
+        billingClientCache = builder.setListener(this).build().also {
+            
+            it.startConnection(
+                object : BillingClientStateListener {
+                    override fun onBillingSetupFinished(billingResult: BillingResult) {
+                        if (!isValidResult(billingResult, promise)) return
 
-        billingClientCache?.startConnection(
-            object : BillingClientStateListener {
-                override fun onBillingSetupFinished(billingResult: BillingResult) {
-                    if (!isValidResult(billingResult, promise)) return
-
-                    promise.safeResolve(true)
-                }
+                        promise.safeResolve(true)
+                    }
 
-                override fun onBillingServiceDisconnected() {
-                    Log.i(TAG, "Billing service disconnected")
-                }
-            })
+                    override fun onBillingServiceDisconnected() {
+                        Log.i(TAG, "Billing service disconnected")
+                    }
+                })
+        }
```

Comment:
@andresesfm This looks so clean and awesome! Such a great example to use `also` in Kotlin 👍